src/snowflake/snowpark/mock/pandas_util.py

```diff
@@ -0,0 +1,162 @@
+            elif isinstance(plain_data[row_idx][col_idx], pandas.Interval):
+
+                def convert_to_python_obj(obj):
+                    if isinstance(obj, numpy.float_):
+                        return float(obj)
+                    elif isinstance(obj, numpy.int_):
+                        return int(obj)
+                    elif isinstance(obj, pandas.Timestamp):
+                        return int(obj.value / 1000)
+                    else:
+                        raise NotImplementedError(
+                            f"[Local Testing] {type(obj)} within pandas.Interval is not supported."
+                        )
+
+                plain_data[row_idx][col_idx] = {
+                    "left": convert_to_python_obj(plain_data[row_idx][col_idx].left),
+                    "right": convert_to_python_obj(plain_data[row_idx][col_idx].right),
+                }
+            elif isinstance(plain_data[row_idx][col_idx], str):
+                pass
+            elif isinstance(plain_data[row_idx][col_idx], pd.Period):
+                # snowflake returns the ordinal of a period object
+                plain_data[row_idx][col_idx] = plain_data[row_idx][col_idx].ordinal
+
+    fields = []
+    for idx, pandas_type in enumerate(data.dtypes):
+        if isinstance(pandas_type, pd.IntervalDtype):
+            data_type = VariantType()
+        elif isinstance(pandas_type, pd.DatetimeTZDtype):
+            data_type = TimestampType()
+        elif pandas_type.type == numpy.float64:
+            data_type = DoubleType()
+        elif isinstance(pandas_type, (pd.Float32Dtype, pd.Float64Dtype)):
+            data_type = DoubleType()
+        elif (
+            pandas_type.type == numpy.int64
+            or pandas_type.type == numpy.datetime64
+            or pandas_type.type == numpy.timedelta64
+        ):
+            data_type = LongType()
+        elif isinstance(pandas_type, PANDAS_INTEGER_TYPES):
+            data_type = LongType()
+        elif isinstance(pandas_type, pd.PeriodDtype):
+            data_type = LongType()
+        elif pandas_type.type == numpy.bool_:
+            data_type = BooleanType()
+        else:
+            data_type = StringType(length=16777216)
+        struct_field = StructField(col_names[idx], datatype=data_type, nullable=True)
+        fields.append(struct_field)
+
+    return StructType(fields=fields), plain_data
+
+
+def _convert_dataframe_to_table(
+    data: "DataFrame", table_name: str, session: "Session"
+) -> Table:
+    """
+    used by create_dataframe from a pandas dataframe to convert a mocking dataframe into a table
+    """
+    df_select_statement, df_plan = data._select_statement, data._plan
+    table = Table(table_name, session)
+    # the original _select_statement & plan of Table is query table name
+    # replace the table._select_statement & plan with the df mocking one
+    table._select_statement, table._plan = df_select_statement, df_plan
```

Comment:
Does this register the table in `TableRegistry`? Why do we need the return value to be a `Table`?

Reply:
this is the existing snowpark python behavior that create from pandas df returns a table

Reply:
updated to register table